src/parser/jobs/rpr/modules/ImmortalSacrifice.tsx

```diff
@@ -0,0 +1,154 @@
+				startTime: event.timestamp,
+				closed: false,
+				consumed: false,
+				stackCount: 0,
+			}
+
+			this.windows.push(window)
+		}
+
+		window.stackCount = event.data ?? window.stackCount + 1
+	}
+
+	private onPlentifulHarvest(event: Events['action']) {
+		const window = this.getOpenWindow()
+
+		if (window === null) {
+			this.brokenLog.trigger(this, 'PH without immortal sacrifice stack',
+				<Trans id="rpr.immortal-sacrifice.trigger.no-immortal-sacrifice">
+					<DataLink action="PLENTIFUL_HARVEST"/> was executed but there were
+					no <DataLink status="IMMORTAL_SACRIFICE"/> stacks.
+				</Trans>)
+			return
+		}
+
+		this.debug('Closing window because plentiful harvest were used', window)
+
+		window.closed = true
+		window.endTime = event.timestamp
+		window.consumed = true
+	}
+
+	private onImmortalSacrificeExpiration(event: Events['statusRemove']) {
+		const window = this.getOpenWindow()
+
+		// valid case, means it was consumed
+		if (window === null) {
+			return
+		}
+
+		this.debug('Closing window that was expired before consumption', window)
+
+		window.closed = true
+		window.consumed = false
+		window.endTime = event.timestamp
+	}
+
+	private getOpenWindow() {
+		if (this.windows.length > 0) {
+			const w = this.windows[this.windows.length - 1]
+
+			if (!w.closed) {
+				return w
+			}
+
+		}
+
+		return null
+	}
+
+	private onComplete() {
+		this.debug(() => {
+			this.debug('Immortal sacrifice windows')
+
+			this.windows.forEach(w => {
+				const start = this.parser.formatEpochTimestamp(w.startTime)
+				const end = w.endTime != null ? this.parser.formatEpochTimestamp(w.endTime) : '<Not Closed>'
+
+				this.debug(`Start: ${start}, End: ${end}, Closed: ${w.closed}, Consumed: ${w.consumed}, Stack Count: ${w.stackCount}`)
+			})
+		})
+
+		const totalImmortalSacrifices = this.windows.length
+
+		if (totalImmortalSacrifices <= 0) {
+			return
+		}
+
+		const consumedImmortalSacrifices = this.windows.filter(w => w.consumed).length
+		const percentUsed = (consumedImmortalSacrifices / totalImmortalSacrifices * 100).toFixed(2)
+
+		this.checklist.add(new Rule({
+			name: <Trans id="rpr.immortal-sacrifice.checklist.title">Uses all Immortal Sacrifices stack</Trans>,
+			description: <Trans id="rpr.immortal-sacrifice.checklist.description">
+				<DataLink status="IMMORTAL_SACRIFICE"/> stacks will be generated when you
+				use <DataLink action="ARCANE_CIRCLE"/>, you can then consume those stacks by
+				using <DataLink action="PLENTIFUL_HARVEST"/>, not only it's a high potency skill, it also grants enough
+				Shroud gauge for a <DataLink action="ENSHROUD"/> usage.
+			</Trans>,
+			displayOrder: DISPLAY_ORDER.IMMORTAL_SACRIFICE,
+			requirements: [
+				new Requirement({
+					name: <Trans id="rpr.immortal-sacrifice.checklist.requirement.name">
+						<DataLink status="IMMORTAL_SACRIFICE"/> stacks consumed
+					</Trans>,
+					overrideDisplay: `${consumedImmortalSacrifices} / ${totalImmortalSacrifices} (${percentUsed}%)`,
```

Comment:
Not sure why you'd wanna override the display for this, but if you do may be worth forcing decimal places.